internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -524,21 +524,25 @@ func (c *enrollCmd) enrollWithBackoff(ctx context.Context, persistentConfig map[
 	}
 
 	c.log.Infof("1st enrollment attempt failed, retrying enrolling to URL: %s with exponential backoff (init %s, max %s)", c.client.URI(), enrollBackoffInit, enrollBackoffMax)
+
 	signal := make(chan struct{})
 	defer close(signal)
 	backExp := backoff.NewExpBackoff(signal, enrollBackoffInit, enrollBackoffMax)
 
 	for {
 		retry := false
-		if errors.Is(err, fleetapi.ErrTooManyRequests) {
-			c.log.Warn("Too many requests on the remote server, will retry in a moment.")
+		switch {
+		case errors.Is(err, fleetapi.ErrTooManyRequests):
+			c.log.Warnf("%s. Too many requests on the remote server, will retry in a moment.", err.Error())
 			retry = true
-		} else if errors.Is(err, fleetapi.ErrConnRefused) {
-			c.log.Warn("Remote server is not ready to accept connections, will retry in a moment.")
+		case errors.Is(err, fleetapi.ErrConnRefused):
+			c.log.Warnf("%s. Remote server is not ready to accept connections, will retry in a moment.", err.Error())
 			retry = true
-		} else if errors.Is(err, fleetapi.ErrTemporaryServerError) {
-			c.log.Warn("Remote server failed to handle the request, will retry in a moment.")
+		case errors.Is(err, fleetapi.ErrTemporaryServerError):
+			c.log.Warnf("%s. Remote server failed to handle the request, will retry in a moment.", err.Error())
```

Comment:
Ok here we want err.Error() as this error can have multiple reasons; maybe something like "Remote server failed to handle the request (%s), will retry in a moment."?